pkg/api/v20170701/validate.go

```diff
@@ -201,20 +171,6 @@ func (a *Properties) Validate() error {
 	return nil
 }
 
-func validateNameEmpty(name string, label string) error {
-	if name != "" {
-		return fmt.Errorf("%s must be an empty value", label)
-	}
-	return nil
-}
-
-func validateName(name string, label string) error {
-	if name == "" {
-		return fmt.Errorf("%s must be a non-empty value", label)
-	}
-	return nil
-}
-
 func validatePoolName(poolName string) error {
 	// we will cap at length of 12 and all lowercase letters since this makes up the VMName
 	poolNameRegex := `^([a-z][a-z0-9]{0,11})$`
```

Comment:
We can take a look at regex later, when this package get more mature. This is the current state for regex.

```
regex
	a regex validator won't be added because commas and = signs can be part
	of a regex which conflict with the validation definitions. Although
	workarounds can be made, they take away from using pure regex's.
	Furthermore it's quick and dirty but the regex's become harder to
	maintain and are not reusable, so it's as much a programming philosiphy
	as anything.

        In place of this new validator functions should be created; a regex can
	be used within the validator function and even be precompiled for better
	efficiency within regexes.go.

	And the best reason, you can submit a pull request and we can keep on
	adding to the validation library of this package!
```